prog/vnet/update_load_balancer.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::UpdateLoadBalancer < Prog::Base
+  subject_is :vm
+
+  def load_balancer
+    @load_balancer ||= LoadBalancer[frame.fetch("load_balancer_id")]
+  end
+
+  label def update_load_balancer
+    vm.vm_host.sshable.cmd("sudo ip netns exec #{vm.inhost_name} nft --file -", stdin: generate_lb_based_nat_rules)
+
+    pop "load balancer is updated"
+  end
+
+  label def remove_load_balancer
+    vm.vm_host.sshable.cmd("sudo ip netns exec #{vm.inhost_name} nft --file -", stdin: generate_nat_rules(vm.ephemeral_net4.to_s, vm.nics.first.private_ipv4.network.to_s))
+    pop "load balancer is updated"
+  end
+
+  def generate_lb_based_nat_rules
+    public_ipv4 = vm.ephemeral_net4.to_s
+    public_ipv6 = vm.ephemeral_net6.to_s
+    private_ipv4 = vm.nics.first.private_ipv4.network
+    private_ipv6 = vm.nics.first.private_ipv6.nth(2)
+    neighbor_vms = load_balancer.vms.reject { _1.id == vm.id }
+    neighbor_ips_v4_set, neighbor_ips_v6_set = generate_lb_ip_set_definition(neighbor_vms)
+    modulo = load_balancer.vms.count
+    ipv4_map_def, ipv6_map_def = generate_lb_map_defs
+
+    balance_mode_ip4, balance_mode_ip6 = if load_balancer.algorithm == "round_robin"
+      ["numgen inc", "numgen inc"]
+    elsif load_balancer.algorithm == "source_hash"
+      ["jhash ip saddr . tcp sport . ip daddr . tcp dport", "jhash ip6 saddr . tcp sport . ip6 daddr . tcp dport"]
+    else
+      raise "Unsupported load balancer algorithm: #{load_balancer.algorithm}"
```

Comment:
`fail` instead of `raise`, see https://github.com/ubicloud/ubicloud/commit/aa6efb1bcfa20fb5e045ff241b2106f7891fe596